web/modules/custom/dbcdk_community/src/Content/FlaggableContentRepository.php

```diff
@@ -182,4 +204,47 @@ public function getContentByIdAllFlags($flag_id) {
     return $flagged_content;
   }
 
+  /**
+   * Get collections for a FlaggableContent object.
+   *
+   * This method provides a simple way to fetch collections for a
+   * FlaggableContent object.
+   *
+   * @param \Drupal\dbcdk_community\Content\FlaggableContent $content
+   *   The FlaggableContent object we wish to fetch collections from.
+   * @param string $type
+   *   The type of collection we wish to fetch.
+   *
+   * @return \DBCDK\CommunityServices\Model\ImageCollection[]|\DBCDK\CommunityServices\Model\VideoCollection[]
+   *   On array of collections for the provided FlaggableContent object.
+   */
+  public function getCollections(FlaggableContent $content, $type = 'image') {
```

Comment:
I do not think it is a good practice to let a method return multiple types if they do not have a shared interface or parent class.

I understand that there is complexity here but perhaps this should be changed to 2 public methods and a shared protected utility method?